src/core/operator_list.js

```diff
@@ -701,11 +670,12 @@ class OperatorList {
             PDFJSDev.test("!PRODUCTION || TESTING")
           ) {
             assert(
-              arg.data instanceof Uint8ClampedArray,
+              arg.data instanceof Uint8ClampedArray ||
+                typeof arg.data === "string",
               'OperatorList._transfers: Unsupported "arg.data" type.'
             );
           }
-          if (!arg.cached) {
+          if (!arg.cached && arg.data && arg.data.buffer) {
```

Comment:
Nit: Maybe `if (!arg.cached && arg.data && arg.data.buffer instanceof ArrayBuffer) {` to prevent *any possible* future issues (since `arg.data` will be a string in some cases)?